### ISSUE TITLE
[FE] Direct Deposit: Edit mode is not exited when moving between Direct Deposit and other Profile sections

### ISSUE THREAD

**Issue by erikphansen** [department-of-veterans-affairs/va.gov-team#25788]
1. Sign into staging with user 378
2. Go to https://staging.va.gov/profile/direct-deposit
3. Start editing bank info
4. Navigate to another page _within_ the Profile
5. Return to the Direct deposit information section
6. Notice that we are still in "edit" mode

https://user-images.githubusercontent.com/20728956/121097762-62727200-c7a9-11eb-81a4-bf05888a9c69.mp4

The same thing happens with contact info as well.

This does not happen when you navigate to a page _outside_ of the Profile and come back since that triggers a full page refresh.

## Expected behavior
No fields should be in edit mode when you first load any section of the Profile

## Possible fix
Exit edit mode when the component in question "unmounts." We could likely accomplish this using the "cleanup" function returned by a `useEffect` hook.

**Comment by erikphansen** [comment]
@andaleliz Basically we're just wondering if you think that this behavior is acceptable or not. If it's fine that bank info and contact info fields remain in "edit" mode as you move throughout the Profile, we can close this. But that behavior felt weird to me.

**Comment by andaleliz** [comment]
@erikphansen, yeah I agree that it feels weird, but at the same time it is probably "acceptable" since a user would still get a warning if they had made changes in the field and then tried to navigate to another section of profile.

We can do better than acceptable though! I think it would feel like all loose ends are tied up if we reset the section back to read mode if someone tries to navigate away.  

My instinct is to do it "quietly" for sighted users - if they open to edit, don't enter anything in the inputs, and navigate away, to me that shows intention that they don't really want to make changes there. So I think it's okay to just reset it without an alert and let them go about their business.  The thing I'm not certain about is how to best handle this for screenreaders. 

@joshkimux and @AngelaFowler82 what are your thoughts on this? Do we need to announce that the section is going back to read mode as the user tries to navigate to a different part of profile? I think the issue description provides a good synopsis of the problem but happy to jump on a call if you need more info.

**Comment by joshkimux** [comment]
@andaleliz I don't think we need to announce that to folks as resetting back to read mode is expected behavior when switching between pages.

**Comment by AngelaFowler82** [comment]
Also, JAWS and NVDA both make a sound when switching to and from reading mode, making an announcement redundant in those cases.

**Comment by andaleliz** [comment]
Thanks @joshkimux and @AngelaFowler82 for weighing in! 

@erikphansen do you need anything from me to implement this update, to reset back to read mode if someone navigates from edit mode without interacting w/ an input?

**Comment by erikphansen** [comment]
@andaleliz Nope, I just needed confirmation that we should exit edit mode in this particular situations. So we're good to go.

**Comment by andaleliz** [comment]
Assigning this back to @erikphansen since nothing else is needed form me at this time.

**Comment by tmitchellgcio** [comment]
Waiting on PR

**Comment by tmitchellgcio** [comment]
@ajakabcin once this is on staging it should be ready for validation

**Comment by ajakabcin** [comment]
Confirmed - all good on staging. The page is returned to read mode when the user navigates to another page in profile and returns to DD info page (unless they've started to enter information, but that's a separate issue documented in [25792](https://github.com/department-of-veterans-affairs/va.gov-team/issues/25792))
